### PR TITLE
feat: remove bitwise shift operators

### LINKED ISSUE #5397: Remove syntax for bitwise operations in favor of functions?
Do we want to demote bitwise operations, e.g. `>>>` to normal functions? Dropping their explicit syntax. 
I think this would be a reasonable simplification. I don't see why a modern primarily functional language would have explicit syntax for these.

### PR DESCRIPTION
**Pull request by dignissimus**
Resolves #5397

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Overall approach looks good. 👍 Let me know when you want some more feedback.

File: examples/larger-examples/program-analysis/domains/Constant.flix [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import BitwiseShl
```
[the remark this answers is not in the document]
Comment (dignissimus) [reply examples/larger-examples/program-analysis/domains/Constant.flix]: I've now re-written the examples where I use BitwiseShl and BitwiseShr (fde073a155869953ae91d22cd839df7cfb0dd42b), I think the example file might still have issues - I wasn't able to get it to compile using `java -jar flix.jar examples/larger-examples/program-analysis/domains/Parity.flix` but it seems unrelated to these changes so I think I might be able to leave the file as it is now

**Comment by dignissimus** [comment]
I've just finished fixing up the output from the script I wrote to re-write the tests and I'm ready for any further feedback

**Comment by magnus-madsen** [comment]
@dignissimus Excellent. Ill try to take a look when I am home.

File: main/src/library/MutDeque.flix [hunk 2]
```diff
@@ -279,7 +279,7 @@ namespace MutDeque {
     def grow!(d: MutDeque[a, r]): Unit \ { Read(r), Write(r) } =
         let MutDeque(r, a, f, b) = d;
         let c = capacity(d);
-        let arr = Array.new(r, c <<< 1);                  // Allocate new array `arr` with double the capacity of `a`.
+        let arr = Array.new(r, Int32.leftShift(c, 1));                  // Allocate new array `arr` with double the capacity of `a`.
```
Comment (magnus-madsen) [inline main/src/library/MutDeque.flix:282]: Alignment was lost.

File: main/src/library/MutDeque.flix [hunk 3]
```diff
@@ -305,7 +305,7 @@ namespace MutDeque {
         let mc = minCapacity();
         let c = capacity(d);
         if (c > mc) {                                         // Prevent the backing array from shrinking below `minCapacity`.
-            let arr = Array.new(r, c >>> 1);                  // Allocate new array `arr` with half the capacity of `a`.
```
Comment (magnus-madsen) [inline main/src/library/MutDeque.flix:308]: Same here.

File: main/src/library/Int64.flix [hunk 4]
```diff
@@ -272,6 +272,17 @@ namespace Int64 {
             highestOneBitPosition(x) |> Int32.toInt64
         }
 
+    ///
+    /// Returns `x` with the bits shifted right by `y` places
+    ///
+    pub def rightShift(x: Int64, y: Int32): Int64 = $INT64_SHR$(x, y)
```
Comment (magnus-madsen) [inline main/src/library/Int64.flix:278]: In other files you have leftShift first.

**Comment by magnus-madsen** [comment]
Looks good. Almost there 👍

File: main/test/flix/Test.Exp.Binary.Bitwise.flix [hunk 5]
```diff
@@ -216,145 +216,4 @@ namespace Test/Exp/Binary/Bitwise {
     @test
     def testBinaryBitwiseXor24(): Bool = (-1ii ^^^ -1ii) == 0ii
 
-    @test
```
[the remark this answers is not in the document]
Comment (dignissimus) [reply main/test/flix/Test.Exp.Binary.Bitwise.flix:219]: I moved these tests into TestInt8.flix, TestInt16.flix, TestInt32.flix, TestInt64.flix and TestBigInt.flix

File: main/src/ca/uwaterloo/flix/language/dbg/PrettyPrinter.scala [hunk 6]
```diff
@@ -457,8 +457,7 @@ object PrettyPrinter {
       case BinaryOperator.BitwiseAnd => "&&&"
       case BinaryOperator.BitwiseOr => "|||"
       case BinaryOperator.BitwiseXor => "^^^"
-      case BinaryOperator.BitwiseLeftShift => "<<<"
-      case BinaryOperator.BitwiseRightShift => ">>>"
+      case _ => "???"
```
[the remark this answers is not in the document]
Comment (dignissimus) [reply main/src/ca/uwaterloo/flix/language/dbg/PrettyPrinter.scala:460]: Maybe not, should I throw an error here, keep it as "<<<" and ">>>" or do something else?

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/dbg/PrettyPrinter.scala:460]: Let keep it as is.

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/dbg/PrettyPrinter.scala:460]: What I mean is, lets turn it back to:

```
      case BinaryOperator.BitwiseLeftShift => "<<<"
      case BinaryOperator.BitwiseRightShift => ">>>"
```

so we dont get unexpected crashes.

**Comment by magnus-madsen** [comment]
Perfect. Thanks!

**Pull request merged**